src/rules/export.js

```diff
@@ -50,7 +50,13 @@ module.exports = function (context) {
         return
       }
       let any = false
-      remoteExports.forEach((v, name) => (any = true) && addNamed(name, node))
+      remoteExports.forEach((v, name) => (
+        (any = true) &&
```

Comment:
This is what I ended up with, here: https://github.com/benmosher/eslint-plugin-import/pull/332/files#diff-71778249fbd501cbc801fa0ee1cac9a2R53

`default` shouldn't count, so it does trigger the `no exported names` report if you `export * from 'foo'` where `foo` has only a `default` export.